### DISCUSSION TITLE
Is the roadmap up-to-date?

### DISCUSSION THREAD

**Discussion by graydwarf** [mbrlabs/Lorien#85] [General]
Considering some feature ideas/enhancements and wondering what's on the roadmap. Looks like some things are pending to 4.0 and some things might already be done or partially done. Is it up-to-date?

**Comment by mbrlabs** [comment]
No, i didn't update the roadmap since i open sourced the project. What features are you thinking about?

**Reply by graydwarf** [reply]
High level idea without looking into feasibility yet. To make Lorien more useable, I would want ability to add notes to my drawings. I currently use PowerPoint for UI mockups and I use a notes pane to the right of every slide (mockup). See attached image for an example of this using an old mockup. I'd like something similar in Lorien so I don't have to chicken scratch my notes using the mouse. 

- Add notes button to toolbar which shows a notes panel at right. 
- On entity select (may add click support in future), the notes pane updates to show any related notes or blank if none exist.
- The notes pane contains is a RichTextLabel, is horizontally expandable and scroll/grows vertically as needed. 
- The notes pane contains basic font/color controls at top along with a hide button.
- If no entity is selected, it shows notes for the overall tab.
- Multiselecting entities will auto-group them with some form of unique ID system. This might lead into a grouping feature.
- Notes are stored based on unique entity/group ID
- Would have to review how things are saved but assume this will be trivial outside of any security concerns.
- Concerns: Need to think through how a scribble with a note will appear or highlight (click or hover) so that a user knows it has a note attached.

If notes by entity/group is a massive undertaking, I might look at adding a Note stamper system instead. I could then "stamp" as many notes buttons as I need per tab. Basically a prefab box with an N in it that supports pressed events. Each one produces a unique notes id used to reference the notes. These can be placed next to things I want to write notes about. Not as sexy as the first approach but certainly easier. When notes pane is hidden, so are the Note buttons (I think).

![Example](https://user-images.githubusercontent.com/6061635/141811336-c532714e-58d1-4603-89a0-f80576a8aec2.png)

**Comment by graydwarf** [comment]
Another option to simplify is just a single note per tab vs the many to 1 ideas listed above and this might be a better starting approach for this feature. 
- Notes tab can sit at the same level as the tab container and overlay it.
- Notes button on toolbar for toggling visibility of notes pane
- Navigating between tabs will auto-load notes for that tab whether visible or not.
- Notes will auto save on keyboard input and perhaps on some focus changes and close events.

**Comment by YoruNocte** [comment]
> No, i didn't update the roadmap since i open sourced the project. What features are you thinking about?

A PNG export would be nice
